File: est-gts/condor/scripts/5_subset_decomp.jl
include("/mnt/dv/wid/projects4/SolisLemus-network-merging/InPhyNet-Simulations/helpers/helpers.jl")


ntaxa = parse(Int64, ARGS[1])
rep = parse(Int64, ARGS[2])
ils = ARGS[3]
ngt = parse(Int64, ARGS[4])
m = parse(Int64, ARGS[5])


astral_file = "/mnt/dv/wid/projects4/SolisLemus-network-merging/InPhyNet-Simulations/est-gts/data/astral/"
astral_file *= "n$(ntaxa)-r$(rep)-$(ils)-$(ngt)gt-m$(m).treefile"

est_gt_file = "/mnt/dv/wid/projects4/SolisLemus-network-merging/InPhyNet-Simulations/est-gts/data/est-gts/"
est_gt_file *= "n$(ntaxa)-r$(rep)-$(ils)-$(ngt)gt-m$(m).treefile"

subset_folder = "/mnt/dv/wid/projects4/SolisLemus-network-merging/InPhyNet-Simulations/est-gts/data/subsets/"
subset_folder *= "n$(ntaxa)-r$(rep)-$(ils)-$(ngt)gt-m$(m)"

@info "Reading trees"
astral_tree = readTopology(astral_file)
est_gts = readMultiTopology(est_gt_file)

@info "Decomposing"
subsets = sateIdecomp(astral_tree, m)

for (i, subset) in enumerate(subsets)
    @info "$(i)/$(length(subsets))"
    iter_folder = "$(subset_folder)-subset$(i)/"

    if isdir(iter_folder) && isfile(joinpath(iter_folder, "pruned_gts.treefile")) && length(readlines(joinpath(iter_folder, "pruned_gts.treefile"))) == length(est_gts)
        printstyled("\t] ALREADY COMPLETED\n", color=:red)
        continue
    end

    tre0 = pruneTruthFromDecomp(astral_tree, subset)
    iter_gts = [pruneTruthFromDecomp(gt, subset) for gt in est_gts]
    
    if !isdir(iter_folder) mkdir(iter_folder) end
    writeTopology(tre0, joinpath(iter_folder, "tre0.treefile"))
    writeMultiTopology(iter_gts, joinpath(iter_folder, "pruned_gts.treefile"))
end
open("$(subset_folder).nsubsets", "w+") do f
    for j = 1:length(subsets)
        for run_idx = 1:10
            write(f, "$(j),$(run_idx)\n")
        end
    end
end

printstyled("[COMPLETE] ", color = :green)
println("n$(ntaxa)-r$(rep)-$(ils)-$(ngt)gt-m$(m), $(length(subsets)) subsets")

File: est-gts/condor/scripts/5_subset_decomp.R
args <- commandArgs(trailingOnly = TRUE)
ntaxa <- as.integer(args[1])
rep <- as.integer(args[2])
ils <- args[3]
ngt <- as.integer(args[4])
m <- as.integer(args[5])

# File paths
astral_file <- "/mnt/dv/wid/projects4/SolisLemus-network-merging/InPhyNet-Simulations/est-gts/data/astral/"
astral_file <- paste0(astral_file, "n$(ntaxa)-r$(rep)-$(ils)-$(ngt)gt-m$(m).treefile")
est_gt_file <- "/mnt/dv/wid/projects4/SolisLemus-network-merging/InPhyNet-Simulations/est-gts/data/est-gts/"
est_gt_file <- paste0(est_gt_file, "n$(ntaxa)-r$(rep)-$(ils)-$(ngt)gt-m$(m).treefile")
subset_folder <- "/mnt/dv/wid/projects4/SolisLemus-network-merging/InPhyNet-Simulations/est-gts/data/subsets/"
subset_folder <- paste0(subset_folder, "n$(ntaxa)-r$(rep)-$(ils)-$(ngt)gt-m$(m)")

# If we've already done this iteration, quit
quit()
if(file.exists(paste0(subset_folder, ".nsubset"))) {
    quit()
}

# Load packages
library(ape)
library(MSCquartets)

# Load 